modyn/trainer_server/internal/dataset/key_sources/local_key_source.py

```diff
@@ -0,0 +1,62 @@
+import os
+from typing import Optional
+
+from modyn.common.trigger_sample import TriggerSampleStorage
+from modyn.trainer_server.internal.dataset.key_sources import AbstractKeySource
+
+LOCAL_STORAGE_FOLDER = "/tmp/.tmp_offline_dataset"
```

Comment:
In addition to the various changes in the code, I have to change the following two files (correct me if I am wrong because I have never done this):
- docker compose: 
``` 
trainer_server:
    restart: on-failure
    depends_on:
      - storage
      - selector
      - model_storage
    build:
      context: .
      dockerfile: docker/Trainer_Server/Dockerfile
    volumes:
    - downsampling-data:/tmp/offline_dataset # Remove when using a fast other storage directory (line below)
#      - /mnt/ssd/offline_dataset:/tmp/offline_dataset
#      - .:/modyn_host
 ```
- modyn config:
 ```
trainer_server:
  hostname: "trainer_server"
  port: "50057"
  ftp_port: "50058"
  offline_dataset_directory: "/tmp/offline_dataset"
 ```

Reply:
And also updating training server's Dockerfile with:
```
RUN mkdir -p /tmp/offline_dataset
RUN chown appuser /tmp/offline_dataset
```

Reply:
Yes, this looks exactly what like needs to be done!